app/src/main/java/org/mozilla/fenix/settings/creditcards/view/CreditCardEditorView.kt

```diff
@@ -68,20 +87,35 @@ class CreditCardEditorView(
                 val fields = NewCreditCardFields(
                     billingName = name_on_card_input.text.toString(),
                     plaintextCardNumber = CreditCardNumber.Plaintext(cardNumber),
-                    cardNumberLast4 = cardNumber.substring(cardNumber.length - 4),
+                    cardNumberLast4 = cardNumber.last4Digits(),
                     expiryMonth = (expiry_month_drop_down.selectedItemPosition + 1).toLong(),
                     expiryYear = expiry_year_drop_down.selectedItem.toString().toLong(),
                     cardType = CARD_TYPE_PLACEHOLDER
                 )
                 interactor.onSaveCreditCard(fields)
             }
         }
+    }
 
-        card_number_input.text = state.cardNumber.toEditable()
-        name_on_card_input.text = state.billingName.toEditable()
+    /**
+     * Validates the credit card information entered by user
+     */
+    @VisibleForTesting
+    internal fun validateCreditCardInfo(): Boolean {
+        var validEntry = true
+
+        // validate credit card number
+        if (!card_number_input.text.toString().validateCreditCardNumber()) {
+            card_number_layout.error =
+                containerView.context.getString(R.string.credit_card_number_validation_error_message)
+            card_number_title.setTextColor(containerView.context.getColorFromAttr(R.attr.destructive))
```

Comment:
I didn't look too deeply into this. Can you also check the edit login to see if this matches? We're mostly copying from the edit login styles. For instance, we aren't using all caps in the title even though they're in the design because some part of the designs were taken as an experiment. Can you also post screenshots?